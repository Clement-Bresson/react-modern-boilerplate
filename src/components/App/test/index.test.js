// Jest documentation https://facebook.github.io/jest/docs/en/getting-started.html
// Enzyme documentation http://airbnb.io/enzyme/docs/api/

import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { App } from '../'

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders right - Snapshot example', () => {
    const component = shallow(<App />)
    expect(component).toMatchSnapshot()
  })
})
