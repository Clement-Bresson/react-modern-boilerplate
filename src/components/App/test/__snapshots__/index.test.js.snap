// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders right - Snapshot example 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    clickCount={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h1>
          Modern React boilerplate
        </h1>,
        <p>
          React 16, Redux, Storybook 3, Jest & Enzyme, Webpack 3, Babel, Eslint, Styled-components
        </p>,
        <p>
          Example Redux click counter: 
          0
        </p>,
        <p>
          <a
            href="https://github.com/Clement-Bresson/react-modern-boilerplate"
            target="_blank"
          >
            View on Github
          </a>
        </p>,
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Modern React boilerplate",
        },
        "ref": null,
        "rendered": "Modern React boilerplate",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "React 16, Redux, Storybook 3, Jest & Enzyme, Webpack 3, Babel, Eslint, Styled-components",
        },
        "ref": null,
        "rendered": "React 16, Redux, Storybook 3, Jest & Enzyme, Webpack 3, Babel, Eslint, Styled-components",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Example Redux click counter: ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Example Redux click counter: ",
          0,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="https://github.com/Clement-Bresson/react-modern-boilerplate"
            target="_blank"
          >
            View on Github
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View on Github",
            "href": "https://github.com/Clement-Bresson/react-modern-boilerplate",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "View on Github",
          "type": "a",
        },
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h1>
            Modern React boilerplate
          </h1>,
          <p>
            React 16, Redux, Storybook 3, Jest & Enzyme, Webpack 3, Babel, Eslint, Styled-components
          </p>,
          <p>
            Example Redux click counter: 
            0
          </p>,
          <p>
            <a
              href="https://github.com/Clement-Bresson/react-modern-boilerplate"
              target="_blank"
            >
              View on Github
            </a>
          </p>,
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Modern React boilerplate",
          },
          "ref": null,
          "rendered": "Modern React boilerplate",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "React 16, Redux, Storybook 3, Jest & Enzyme, Webpack 3, Babel, Eslint, Styled-components",
          },
          "ref": null,
          "rendered": "React 16, Redux, Storybook 3, Jest & Enzyme, Webpack 3, Babel, Eslint, Styled-components",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Example Redux click counter: ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Example Redux click counter: ",
            0,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="https://github.com/Clement-Bresson/react-modern-boilerplate"
              target="_blank"
            >
              View on Github
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "View on Github",
              "href": "https://github.com/Clement-Bresson/react-modern-boilerplate",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "View on Github",
            "type": "a",
          },
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
